src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1212,6 +1388,161 @@ TupleToShardInterval(HeapTuple heapTuple, TupleDesc tupleDescriptor, Oid interva
 
 
 /*
+ * ReadWorkerNodes iterates over pg_dist_node table, converts each row
+ * into it's memory representation (i.e., WorkerNode) and adds them into
+ * a list. Lastly, the list is returned to the caller.
+ */
+static List *
+ReadWorkerNodes()
```

Comment:
metadata_cache.c is getting needlessly crowded, can we move these functions into their own file?